src/System.Buffers.Experimental/ReferenceCountedBuffer.cs

```diff
@@ -17,15 +17,16 @@ public override void Retain()
             Interlocked.Increment(ref _referenceCount);
         }
 
-        public override void Release()
+        public override bool Release()
         {
             if (!IsRetained) BuffersExperimentalThrowHelper.ThrowInvalidOperationException();
             if (Interlocked.Decrement(ref _referenceCount) <= 0) {
                 OnNoReferences();
             }
+            return IsRetained;
         }
 
-        public override bool IsRetained => _referenceCount > 0;
+        protected override bool IsRetained => _referenceCount > 0;
```

Comment:
@davidfowl, what do you mean by that? Are you proposing that we return the refcount instead of bool? If yes, we cannot do it. There are some lifetime management strategies/algorithms that don't relly know the exact refcount; they just know if the refcount is positive or zero.